libcudacxx/include/cuda/std/__cuda/ensure_current_device.h

```diff
@@ -52,7 +52,7 @@ struct __ensure_current_device
 
   //! @brief If the \p __original_device was not equal to \p __target_device sets the current device back to
   //! \p __original_device
-  ~__ensure_current_device()
+  ~__ensure_current_device() noexcept(false)
```

Comment:
I once tried to also disable `noexcept` on a scope guard. It's a very bad idea, because the crashes from `std::terminate` in case this dtor throws during unwinding are terrible to debug.

I strongly recommend to leave the dtor `noexcpt` and change `_CCCL_TRY_CUDA_API` to either the asserting or logging macro.

Reply:
Would second that